grafana-plugin/src/containers/ColumnsSelector/ColumnsSelector.tsx

```diff
@@ -193,7 +210,10 @@ export const ColumnsSelector: React.FC<ColumnsSelectorProps> = observer(
             tooltipPlacement="top"
             tooltip={'Reset table to default columns'}
             disabled={isResetLoading || isDefaultColumnOrder}
-            onClick={WrapAutoLoadingState(onReset, ActionKey.RESET_COLUMNS_FROM_ALERT_GROUP)}
+            onClick={WrapAutoLoadingState(
+              WrapWithGlobalNotification(onReset, { success: 'Columns list has been reset' }),
```

Comment:
I don't think using `WrapAutoLoadingState` and `WrapWithGlobalNotification` in components directly is a good thing. I can see why we're doing it here: because in components we have functions that wrap multiple store methods and we want loading state / global notification to work with both them at once. 

However, such pattern is a cause of making our containers/components huge. Ideally, we should move towards making our containers and components smaller and less responsible by moving such things into the store layer and encapsulating as much frontend business logic as possible. In this case we could have a dedicated action in MobX store that performs 2 other actions and this wrapper action would be decorated with `@AutoLoadingState` and `@WithGlobalNotification`. Then in component you simply trigger it. And the same could be done with 2 other places in PR where we use `WrapAutoLoadingState(
              WrapWithGlobalNotification(...))`
              
I'm happy to discuss this topic if you want :)

Reply:
I see your point but this would be only possible if I'd move the reset out of the class to the store, but we don't really divide the logic this way in other parts of the app. Reset function handles these 2 calls

```jsx
await alertGroupStore.resetTableSettings();
await alertGroupStore.fetchTableSettings();
```

These together are *only* used in this component, I'd kinda dislike seeing it in the store just so that I can wrap it with the decorators...

Reply:
But also `resetTableSettings` is only used by this component so why not to invoke `fetchTableSettings` within `resetTableSettings` in store?

The idea is to flatten our components and make our view layer simple and dumb:)